### PR TITLE
AAP-17700: Updates to Builder documentation 

### PR DESCRIPTION
**Pull request by hherbly**
This PR updates content on Builder in chapter 2 of Creating and Consuming Execution Environments. See [AAP-17700](https://issues.redhat.com/browse/AAP-17700) for more info.

### CODE REVIEW

**Comment by nitzmahone** [comment]
I notice you've got this labeled `Needs backport to AAP 2.2/2.3`- this content is only applicable to Builder 3.0+, so AAP 2.4+.

File: downstream/modules/builder/con-container_file.adoc [hunk 1]
```diff
@@ -2,13 +2,12 @@
 
 = Containerfile
 
-Once your definition file is created, Ansible Builder reads and validates it, then creates a `Containerfile`, and finally passes the `Containerfile` to Podman to package and create your automation execution environment image using the following instructions:
+Once your definition file is created, {Builder} reads and validates it, creates a `Containerfile`, and finally passes the `Containerfile` to Podman to package and create your automation execution environment image using the following instructions:
```
Comment (nitzmahone) [inline downstream/modules/builder/con-container_file.adoc:5]:
```suggestion
Once your definition file is created, {Builder} reads and validates it, creates a `Containerfile` and container build context, and optionally passes these to Podman to build your automation execution environment image. The container build occurs in several distinct stages: `base` , `galaxy`, `builder`, and `final`. The image build steps (along with any corresponding custom `prepend_` and `append_` steps defined in `additional_build_steps` are:
```

File: downstream/modules/builder/con-container_file.adoc [hunk 2]
```diff
@@ -2,13 +2,12 @@
...
 
 . Fetch base image.
 . In the ephemeral copy of base image, collections are downloaded and the list of declared Python and system dependencies, if any, are collected for later.
 . In the ephemeral builder image, Python wheels for all Python dependencies listed in the definition file are downloaded and built (as needed), including all Python dependencies declared by collections listed in the definition file.
```
Comment (nitzmahone) [inline downstream/modules/builder/con-container_file.adoc:9]:
```suggestion
. During the `builder` build stage, Python dependencies declared by collections are merged with those listed in the definition file. This final set of Python dependencies is downloaded and built as Python wheels and stored for later installation during the `final` build stage.
```

File: downstream/modules/builder/con-container_file.adoc [hunk 3]
```diff
@@ -2,13 +2,12 @@
...
-. Any `prepend_` commands defined for additional_build_steps from the definition file are run.
 . In the final {ExecEnvName} image, system dependencies listed in the definition file are installed, including all system dependencies declared by collections listed in the definition file.
```
Comment (nitzmahone) [inline downstream/modules/builder/con-container_file.adoc:10]:
```suggestion
. During the `final` build stage, the previously-downloaded collections are installed, along with system packages and any previously-built Python packages that were declared as dependencies by the collections or listed in the definition file.
```

File: downstream/modules/builder/con-container_file.adoc [hunk 4]
```diff
@@ -2,13 +2,12 @@
...
 . In the final {ExecEnvName} image, the downloaded collections are copied and the previously fetched Python dependencies are installed.
```
Comment (nitzmahone) [inline downstream/modules/builder/con-container_file.adoc:11]:
```suggestion
```

File: downstream/modules/builder/con-container_file.adoc [hunk 5]
```diff
@@ -2,13 +2,12 @@
...
-. Any `append_` commands defined for additional_build_steps from the definition file are run.
+. Meanwhile, at each of the build stages above, any `prepend_` commands defined for `additional_build_steps` from the definition file are run.
```
Comment (nitzmahone) [inline downstream/modules/builder/con-container_file.adoc:12]:
```suggestion
```

File: downstream/modules/builder/ref-definition-file-images.adoc [hunk 6]
```diff
@@ -1,8 +1,8 @@
 [id="ref-definition-file-image"]
 
-= images
+= Images
 
-The `images` section of the definition file is a dictionary used to define the base image to be used. Verification of signed container images is supported with the `podman` container runtime.
+The `images` section of the definition file is a dictionary used to define the base image. Verification of signed container images is supported with the `podman` container runtime.
```
Comment (nitzmahone) [inline downstream/modules/builder/ref-definition-file-images.adoc:5]:
```suggestion
The `images` section of the definition file identifies the base image. Verification of signed container images is supported with the `podman` container runtime.
```

**Review by nitzmahone — CHANGES_REQUESTED** [review]
Added some suggestions to make the build stage description a bit more linear and concise- dunno if I succeeded :laughing: 

Overall looks good, thanks!

**Review by nitzmahone — APPROVED** [review]

File: downstream/modules/builder/con-building-definition-file.adoc [hunk 7]
```diff
@@ -2,11 +2,9 @@
...
 
-The following is an example of a version 3 definition file. Each definition file should specify the major version number of the Ansible Builder feature set it uses. If not specified, the default version is 1, and most new features and definition keywords will be unavailable.
-
-// ddacosta - there is a Ansible Builder version 3 Porting Guide in progress that discusses the changes from v1/2 to v3. A link should be included to that document once its published.
+The following is an example of a version 3 definition file. Each definition file must specify the major version number of the Ansible Builder feature set it uses. If not specified, {Builder} defaults to version 1, making most new features and definition keywords unavailable.
```
Comment (dcdacosta) [inline downstream/modules/builder/con-building-definition-file.adoc:7]: Ansible Builder should use the attribute {Builder}

File: downstream/modules/builder/con-container_file.adoc [hunk 8]
```diff
@@ -2,13 +2,10 @@
 
 = Containerfile
 
-Once your definition file is created, Ansible Builder reads and validates it, then creates a `Containerfile`, and finally passes the `Containerfile` to Podman to package and create your automation execution environment image using the following instructions:
+Once your definition file is created, {Builder} reads and validates it, creates a `Containerfile` and container build context, and optionally passes these to Podman to build your {ExecEnvNameSing} image. The container build occurs in several distinct stages: `base` , `galaxy`, `builder`, and `final`. The image build steps (along with any corresponding custom `prepend_` and `append_` steps defined in `additional_build_steps`) are:
```
Comment (dcdacosta) [inline downstream/modules/builder/con-container_file.adoc:5]: Suggest changing "Once" to "After". According to Style Guide: Use once only to mean "one time". Do not use as a conjunction to mean "after" or "when".

File: downstream/modules/builder/con-system-dependencies.adoc [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 = System
 
-The `system` entry in the definition points to a https://docs.opendev.org/opendev/bindep/latest/readme.html[bindep] requirements file or to an inline list of bindep entries, which will install system-level dependencies that are outside of what the collections already include as their dependencies. It can be listed as a relative path from the directory of the {ExecEnvNameSing} definition’s folder, or an absolute path. A minimum expectation is that the collection(s) specify necessary requirements for `[platform:rpm]`.
+The `system` entry in the definition points to a link:https://docs.opendev.org/opendev/bindep/latest/readme.html[bindep] requirements file or to an inline list of bindep entries, which install system-level dependencies that are outside of what the collections already include as their dependencies. It can be listed as a relative path from the directory of the {ExecEnvNameSing} definition's folder, or as an absolute path. At a minimum, the the collection(s) should specify necessary requirements for `[platform:rpm]`.
```
Comment (dcdacosta) [inline downstream/modules/builder/con-system-dependencies.adoc:5]: Suggest changing "should specify necessary" to "must specify necessary". "Should" makes the requirements sound optional and I don't think they are.

File: downstream/modules/builder/con-additional-build-files.adoc [hunk 10]
```diff
@@ -2,16 +2,16 @@
...
 
-You can add any file to the build context directory by referring or copying them to the `additional_build_steps` section of the definition file. The format is a list of dictionary values, each with a `src` and `dest` key and value.
+You can add any external file to the build context directory by referring or copying them to the `additional_build_steps` section of the definition file. The format is a list of dictionary values, each with a `src` and `dest` key and value.
 
-Each list item must be a dictionary containing the following (non-optional) keys:
+Each list item must be a dictionary containing the following required keys:
 
-.`src`
-Specifies the source files to copy into the build context directory. This can be an absolute path (for example, `/home/user/.ansible.cfg`), or a path that is relative to the execution environment file. Relative paths can be glob expressions matching one or more files (for example,  `files/*.cfg`).
+== `src`
```
Comment (dcdacosta) [inline downstream/modules/builder/con-additional-build-files.adoc:9]: This should not be a subheading here but rather a list under the statement "Each list item must be a dictionary containing the following required keys:"

I would either change back to the .`scr` and .`dest` block title format or make it a description list. See 
https://asciidoctor.org/docs/asciidoc-writers-guide/#block-titles
https://asciidoctor.org/docs/asciidoc-writers-guide/#description-lists for more info.

File: downstream/modules/builder/con-system-dependencies.adoc (hunk 9 above)
Comment (hherbly) [reply downstream/modules/builder/con-system-dependencies.adoc:5]: I was wondering about that. Thanks for the clarification!

**Pull request merged**